docs-user/bunny-2.md

```diff
@@ -0,0 +1,78 @@
+
+Consider the optimized profile from the previous case study: [https://perfht.ml/2IqQTH2](https://perfht.ml/2IqQTH2)
+
+![A view of the content and worker stacks. The content thread does work, and is blocked as the worker thread does work, making things effectively synchronous.](./images/bunny-analysis/threads-sync.png)
+
+Looking at a zoomed-in view of the thread stack graph shows how the content process' main thread is drawing the visualization to the screen, then asking the worker for the next draw calls. This effectively makes the two threads synchronous as they block each other taking turns doing the work. The blank spots in the content thread (which contain idle stacks) represent time where the content thread is blocked as it waits for the worker to complete its work.
+
+## The code
+
+In a simplified code example, the iframe that is drawing using the [`CanvasRenderingContext2D`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D API) runs code that looks something like this:
+
+```js
+worker.addEventListener('message', message => {
+  if (message.data.type === 'draw') {
+    requestAnimationFrame(() => {
+      // Draw the current set of rectangles.
+      drawRects(message.data.drawCalls);
+      // Post a message to the worker asking for the next draw calls.
+      worker.postMessage({ type: 'generate-draw-calls' });
+    });
+  }
+});
+```
+
+In this example, the iframe draws to the canvas, and once it's done, it posts a message to the worker asking it generate a new list of the draw calls. While this works, it's not harnessing the fact that the worker can run code in parallel. The fix luckily is simple. Swap the order of the `drawRect` and `worker.postMessage` calls.
+
+```js
+worker.addEventListener('message', message => {
+  if (message.data.type === 'draw') {
+    requestAnimationFrame(() => {
+      // Ask the worker to generate the new draw calls for the NEXT frame.
+      worker.postMessage({ type: 'generate-draw-calls' });
+      // Draw the current frame.
+      drawRects(message.data.drawCalls);
+    });
+  }
+});
+```
+
+## The new profile
+
+Profiling this change reveals the new behavior: https://perfht.ml/2IpSRr4
+
+![A view of the content and worker stacks. The content thread and worker thread do work in parallel, and are not blocked by each other.](./images/bunny-analysis/threads-parallel.png)
+
+Now the content process' main thread is not blocked by the worker. The overall time to draw a single frame is the same, but conceptually this has an important difference. The blank spots in the content thread no longer represent time where it is blocked. Now it truly represents idle time, where no work needs to be done.
+
+## How this improves performance
+
+Consider what happens if the worker takes a long time to compute the new draw calls. As the work scales, it happens in parallel. To simulate this, a function called `doWork` can be added to the worker.
```

Comment:
"takes a long time" => I think you can rework this sentence a little bit, like "Consider what happens if the code to compute the new draw calls is more complex and needs more time". 
My problem is that the sentence "the worker takes a long time" is quite passive, while what *can* happen is that the work needed is more complex.